src/steamship/agents/schema/chathistory.py

```diff
@@ -140,8 +140,22 @@ def append_message_with_role(
             chunk_tags = self.text_splitter.chunk_text_to_tags(
                 text, kind=TagKind.CHAT, name=ChatTag.CHUNK
             )
-            block.tags.extend(chunk_tags)
-            self.embedding_index.insert(chunk_tags)
+            saved_tags = []
```

Comment:
Can the chunk_text_to_tags take care of this? 

Another question: Do we need a text_splitter here? One could assume chat messages are small enough to fit in 1 embedding?

Reply:
Good suggestion, cleaned that up I think.

I added text_splitter because I couldn't answer the question "should we assume chat messages are small enough to fit in an embedding".  To me having them chunked to preserve some sensible embedding performance seemed like a safer bet.